### DISCUSSION TITLE
Creating LIKE parameter that includes multiple letters in each group

### DISCUSSION THREAD

**Discussion by Rikijs** [symfony/symfony#45837] [Q&A]
Hello!

I am having problems with SQL LIKE.

I have 10 input fields.
The first 5 inputs (1-5) represent letters of which word consist of (1 letter for each field).
The second 5 inputs (fields 6-10) represent the letters (one or more in each field) that word do not have (at position 1-5).

For example i use LIKE with word "august" it gets LIKE parameter "aug__" if 3 letters are known and 2 letters can be any letter.
At the same time one can restrict letters that are known to not be at some position. For example "___ze" restricts 4th to not be z and 5th to not be e (for 1 letter in each position).

My problem is - how to construct string `termNo` for LIKE that includes multiple letters (that are not ir word) in each position?

```
public function findLikeComboLength5($termYes, $termNo): mixed
{
    return $this->createQueryBuilder('word')
        ->andWhere('word.name LIKE :termSearchYes')
        ->andWhere('word.name not LIKE :termSearchNo')
        ->andWhere('LENGTH(word.name) = :termLength')
        ->setParameter('termSearchYes', '%'. $termYes .'%')
        ->setParameter('termSearchNo', '%'. $termNo .'%')
        ->setParameter('termLength', 5)
        ->setMaxResults(1000)
        ->getQuery()
        ->execute();
}
```

Showing the path to right solution most welcome!
Thank you

**Comment by stof** [comment]
this looks like a question about SQL rather than a question about Symfony (btw, your code snippet is about Doctrine, not about Symfony). I suggest you to ask this on StackOverflow instead (asking it in term of pure SQL rather than in term of a Doctrine QueryBuilder will give you a broader reach btw)

**Comment by helmis123** [comment]
`SELECT * FROM (SELECT * FROM word WHERE word.name LIKE '%aug%' ) WHERE name NOT LIKE '%ze%' ;`
If it matches what you want, I can write you the DQL query.
